src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstodeliusupdate/services/CourtRegisterSyncService.kt

```diff
@@ -19,41 +20,37 @@ class CourtRegisterSyncService(
     courtRegisterCourts: List<CourtDto>
   ): SyncStatistics {
 
-    // get all active / inactive courts from Probation (DELIUS)
-    val allCourtsHeldInProbation = probationCourts.map { ProbationService.CourtDataToSync(it) }.associateBy { it.code }
-
-    // get all the courts from the register
-    val allRegisteredCourts: MutableList<ProbationService.CourtDataToSync> = mutableListOf()
-    courtRegisterCourts.forEach {
-      allRegisteredCourts.add(ProbationService.CourtDataToSync(it))
-    }
-    val courtRegisterMap = allRegisteredCourts.associateBy { it.code }
+    val probationCourtsMap = probationCourts.map { CourtDataToSync(it) }.associateBy { it.code }
+    val courtRegisterMap = courtRegisterCourts.map { CourtDataToSync(it) }.associateBy { it.code }
 
     val stats = SyncStatistics()
-    // matches
-    courtRegisterMap.filter { c -> allCourtsHeldInProbation[c.key] != null }
+    courtRegisterMap.filterMatching(probationCourtsMap)
       .forEach {
         // TODO("Evaluate the court Register ProbationAreaCode from the postcode or an API")
         // The probationAreaCode should be calculated from the Court Register address or through an API before the sync process
         // Until this is determined, copy over the ProbationAreaCode from the corresponding Probation Court Data
-        val courtFromRegisterWithProbationArea = it.value.copy(probationAreaCode = allCourtsHeldInProbation[it.key]!!.probationAreaCode)
-        courtRegisterUpdateService.syncCourt(allCourtsHeldInProbation[it.key], courtFromRegisterWithProbationArea, stats)
+        val courtFromRegisterWithProbationArea = it.value.copy(probationAreaCode = probationCourtsMap[it.key]!!.probationAreaCode)
+        courtRegisterUpdateService.syncCourt(probationCourtsMap[it.key], courtFromRegisterWithProbationArea, stats)
       }
 
-    // new
-    courtRegisterMap.filter { c -> allCourtsHeldInProbation[c.key] == null }
+    courtRegisterMap.filterNew(probationCourtsMap)
       .forEach { courtRegisterUpdateService.syncCourt(null, it.value, stats) }
 
-    // not there / inactive
-    allCourtsHeldInProbation.filter { c -> c.value.active && courtRegisterMap[c.key] == null }
+    probationCourtsMap.filterInactiveOrMissing(courtRegisterMap)
       .forEach {
         courtRegisterUpdateService.syncCourt(
-          allCourtsHeldInProbation[it.key],
+          probationCourtsMap[it.key],
           it.value.copy(active = false),
           stats
         )
       }
 
     return stats
   }
+
+  private fun <K, V> Map<out K, V>.filterNew(courtsMap: Map<K, V>): Map<K, V> = filter { c -> courtsMap[c.key] == null }
+  private fun <K, V> Map<out K, V>.filterMatching(courtsMap: Map<K, V>): Map<K, V> = filter { c -> courtsMap[c.key] != null }
+  private fun <K, V> Map<out K, CourtDataToSync>.filterInactiveOrMissing(courtsMap: Map<K, V>): Map<K, CourtDataToSync> {
+    return filter { c -> c.value.active && courtsMap[c.key] == null }
+  }
```

Comment:
I wish I could understand this, it looks awesome.

Reply:
I basically pulled out the code that was doing the filtering - the Key/Value pair declarations for Kotlin crowd the methods so I can see it's a bit awkward to understand. 
It's the actual filtering in the {} that's the important bit as it is just filtering the map